src/main/groovy/io/seqera/wave/configuration/SpackConfig.groovy

```diff
@@ -61,7 +61,7 @@ class SpackConfig {
     /**
      * The container image used for Spack builds
      */
-    @Value('${wave.build.spack.builderImage:`spack/ubuntu-jammy:v0.20.0`}')
+    @Value('${wave.build.spack.builderImage:`spack/ubuntu-jammy:develop`}')
```

Comment:
```suggestion
    @Value('${wave.build.spack.builderImage:`cr.seqera.io/spack/wave/spack/ubuntu-jammy:develop`}')
```

I actually just realized that moving to `spack/ubuntu-jammy:develop` won't be useful, since our workflow goes like this:
- a cron job starts every day to build spack packages in our spack stack against `spack/ubuntu-jammy:develop`
- the built packages are pushed to our build cache
- only if the build succeeded, retag `spack/ubuntu-jammy:develop` into `195996028523.dkr.ecr.eu-west-1.amazonaws.com/wave/spack/ubuntu-jammy:develop`, which is proxied by Harbor at the address in the suggestion above

So we should make wave use our `develop` image, not the one from the spack project, because that may break for our spack stack, and the GHA would notify us of that.

Here's the cron job I built to test the GHA that automatically performs the actions above: https://github.com/seqeralabs/test-nf-tower-cloud-gha-dev-branch-pr/blob/master/.github/workflows/populate-spack-binary-mirror.yml